js/libs/ui-shared/src/user-profile/SelectComponent.tsx

```diff
@@ -63,7 +67,13 @@ export const SelectComponent = (props: UserProfileFieldProps) => {
             selections={
               field.value ? field.value : isMultiValue ? [] : t("choose")
             }
-            variant={isMultiValue ? "typeaheadmulti" : "single"}
+            variant={
+              isMultiValue
+                ? "typeaheadmulti"
+                : options.length >= 10
+                  ? "typeahead"
+                  : "single"
+            }
```

Comment:
I don't think we want this here, because this is the base that is used through out the application and we don't want this to be dynamic in all cases